File: equipment-frontend/src/types/equipmentPositionHistory.ts
export interface EquipmentPosition {
    equipmentId: string;
    lat: number;
    lon: number;
}

export interface EquipmentPositionHistory {
    equipmentId: string;
    positions: {
        lat: number;
        lon: number;
    }[];
}

export interface EquipmentStateHistory {
    equipmentId: string;
    states: {
        date: string;
        equipmentStateId: string; // Ajustado para corresponder ao JSON
    }[];
}

File: aiko/Models/EquipmentPositionHistory.cs
using System.Collections.Generic;

namespace aiko.Models
{
    public class EquipmentPositionHistory
    {
        public string EquipmentId { get; set; }
        public List<Position> Positions { get; set; }
    }

    public class Position
    {
        public DateTime Date { get; set; }
        public double Lat { get; set; }
        public double Lon { get; set; }
    }
}
